### PR TITLE
Add a VFS gRPC API + server based on the FUSE API.

### PR DESCRIPTION
**Pull request by vadimberezniker**
---

**Version bump**: None

### CODE REVIEW

File: proto/vfs.proto [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  rpc Write(WriteRequest) returns (WriteResponse);
+  rpc Fsync(FsyncRequest) returns (FsyncResponse);
+  // Flush is called whenever a file descriptor is closed. This may be called
+  // more than once if the file descriptor was duplicated. Errors from this op
+  // are returned to the file system user.
+  rpc Flush(FlushRequest) returns (FlushResponse);
+  //  Release is called when there are no more file descriptor references
+  //  left.
+  //  Any errors returned here are not visible to the file system user.
+  rpc Release(ReleaseRequest) returns (ReleaseResponse);
```
Comment (tylerwilliams) [inline proto/vfs.proto:159]: nit: newlines?

File: proto/vfs.proto [hunk 2]
```diff
@@ -0,0 +1,172 @@
+syntax = "proto3";
```
Comment (tylerwilliams) [inline proto/vfs.proto:1]: Could you add a comment at the top linking the source that this is from / based on?

**Review by tylerwilliams — APPROVED** [review]
this is really cool

File: enterprise/server/util/vfs_server/vfs_server.go [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+	return &Server{
+		env:           env,
+		workspacePath: workspacePath,
+		inputFiles:    make(map[string]*repb.FileNode),
+		fileHandles:   make(map[uint64]*fileHandle),
+		nextHandleID:  1,
+	}
+}
+
+func (p *Server) Prepare(ctx context.Context, fsLayout *container.FileSystemLayout) error {
```
[the remark this answers is not in the document]
Comment (vadimberezniker) [reply enterprise/server/util/vfs_server/vfs_server.go]: Done.

File: proto/vfs.proto (hunk 1 above)
Comment (vadimberezniker) [reply proto/vfs.proto]: Done.

File: proto/vfs.proto (hunk 2 above)
Comment (vadimberezniker) [reply proto/vfs.proto:6]: Done.

**Pull request merged**